src/utils/AuthProvider.tsx

```diff
@@ -97,13 +97,33 @@ export default function AuthProvider({
   // sign up and set the session, userId, and userEmail
   const signUp = useCallback(
     async (email: string, password: string, options: SignUpOptions) => {
-      const value = await supabase.auth.signUp({
+      const { data, error } = await supabase.auth.signUp({
         email,
         password,
         options,
       }); // will trigger onAuthStateChange to update the session
 
-      return value;
+      let authError = null;
+
+      // This code is largely taken from https://github.com/orgs/supabase/discussions/1282#discussioncomment-5230475
+
+      // User exists, but is fake. See https://supabase.com/docs/reference/javascript/auth-signup
+      if (
+        data.user &&
+        data.user.identities &&
+        data.user.identities.length === 0
+      ) {
+        authError = {
```

Comment:
instead of creating `authError`, we could do something like the following to directly update the `value` (`AuthResponse`):

        const authError = new AuthError(
          'A user account with this email already exists',
        );
        value.error = authError;